### PR TITLE
Add batched migration script

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes
Add batched migration script

### CODE REVIEW

File: scripts/SqlServer/Batched/1_Migration_Setup.sql [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    WHERE
+        ej.[Tags] IS NOT NULL
+      AND LEN(ej.[Tags]) > 0
+      AND ej.[Ordering] NOT IN (SELECT t.[ordering_id] FROM [dbo].[tags] t);
+        
+    WHILE @from_id <= @max_id
+    BEGIN
+        BEGIN TRAN;
+        INSERT INTO [dbo].[tags]([ordering_id], [tag])
+        SELECT * FROM (SELECT records.[Ordering], cross_product.[items] FROM (
```
Comment (to11mtm) [inline scripts/SqlServer/Batched/1_Migration_Setup.sql]: Would prefer specific columns over select * here

File: scripts/SqlServer/Batched/1_Migration_Setup.sql [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            SELECT *
+            FROM [dbo].[EventJournal] AS ej
+            WHERE ej.[Ordering] >= @from_id AND ej.[Ordering] <= @from_id + 1000
+        ) AS records CROSS APPLY [dbo].[Split](records.Tags, ';') cross_product) AS s([ordering_id], [tag])
+        WHERE NOT EXISTS (
+                SELECT * FROM [dbo].[tags] t WITH (updlock)
```
Comment (to11mtm) [inline scripts/SqlServer/Batched/1_Migration_Setup.sql:64]: You can probably `SELECT 1` instead of `SELECT *` here.

**Review by to11mtm — COMMENTED** [review]
@Arkatufus I'm looking at these with the mindset that they are expected to be modified for a specific use case.

That aside,

- We should give our function names unique names if we are using `CREATE OR ALTER`, especially in the case of functions named things like `Split` which may have an existing declaration in a Prod DB.
- We should have documentation that makes it abundantly clear to users that these scripts must be modified for their specific table configuration(s).

**Comment by Arkatufus** [comment]
@to11mtm makes sense, I'll try to incorporate those changes

File: scripts/SqlServer/Batched/1_Migration_Setup.sql (hunk 1 above)
Comment (Arkatufus) [reply scripts/SqlServer/Batched/1_Migration_Setup.sql]: The `SELECT *` is actually nescessary here, the specific columns are defined inside the inner `SELECT`

**Comment by Aaronontheweb** [comment]
@Arkatufus needs a rebase

**Comment by Arkatufus** [comment]
Rebased to recent dev branch in #168

**Pull request closed without merging**